File: benchmarks/successor/representation-v1/transformed/deterministic/fsharp/gold/007-query-engine-refactor/Program.fs
open System
open System.Text.Json

let options =
    JsonSerializerOptions(
        PropertyNameCaseInsensitive = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
    )

let mutable running = true

while running do
    match Console.ReadLine() with
    | null -> running <- false
    | loc_ey5euhrnbl ->
        try
            let loc_zk64hdzcrs =
                JsonSerializer.Deserialize<OrderFlowEngine.Request>(loc_ey5euhrnbl, options)

            match loc_zk64hdzcrs with
            | null -> raise (InvalidOperationException("Request was null"))
            | loc_zk64hdzcrs ->
                let loc_r43bavyf4z = OrderFlowEngine.handle loc_zk64hdzcrs
                Console.WriteLine(JsonSerializer.Serialize(loc_r43bavyf4z, options))
        with loc_fnvfrnyz4a ->
            Console.WriteLine(JsonSerializer.Serialize({| error = loc_fnvfrnyz4a.Message |}, options))

File: benchmarks/successor/representation-v1/transformed/deterministic/csharp/gold/007-query-engine-refactor/Program.cs
using System.Text.Json;

namespace OrderFlow;

public static class Program
{
    private static readonly JsonSerializerOptions Options = new()
    {
        PropertyNameCaseInsensitive = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
    };

    public static void Main()
    {
        string? loc_ey5euhrnbl;
        while ((loc_ey5euhrnbl = Console.ReadLine()) is not null)
        {
            try
            {
                var loc_zk64hdzcrs = JsonSerializer.Deserialize<Request>(loc_ey5euhrnbl, Options)
                    ?? throw new InvalidOperationException("Request was null");
                var loc_r43bavyf4z = OrderFlowEngine.Handle(loc_zk64hdzcrs);
                Console.WriteLine(JsonSerializer.Serialize(loc_r43bavyf4z, Options));
            }
            catch (Exception loc_fnvfrnyz4a)
            {
                Console.WriteLine(JsonSerializer.Serialize(new { error = loc_fnvfrnyz4a.Message }, Options));
            }
        }
    }
}
